### PR TITLE
fix: Read more link doesn't show more content for release notes

### PR DESCRIPTION
**Pull request by tsl143**
Fixes: #3148 

**Before:** 

screen shot 2017-09-17 at 4 45 36 pm

**After:**

screen shot 2017-09-17 at 4 44 37 pm

### CODE REVIEW

File: tests/unit/ui/components/TestShowMoreCard.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ describe(__filename, () => {
 
   it('truncates the contents if they are too long', () => {
     const root = render({ children: 'Hello I am description' });
-    root.truncateToMaxHeight({ clientHeight: 101 });
+    root.truncateToMaxHeight({ clientHeight: 146 });
```
Comment (willdurand) [inline tests/unit/ui/components/TestShowMoreCard.js]: Could you export `MAX_HEIGHT` in the `ShowMoreCard` component and use it here? Like:

```
root.truncateToMaxHeight({ clientHeight: MAX_HEIGHT + 1 });
```

File: src/ui/components/ShowMoreCard/index.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ import Card from 'ui/components/Card';
 import './ShowMoreCard.scss';
 
 
-const MAX_HEIGHT = 100;
+const MAX_HEIGHT = 145;
```
Comment (willdurand) [inline src/ui/components/ShowMoreCard/index.js]: We should maybe add a comment saying that it is related to the CSS rule (if I remember correctly).

File: tests/unit/ui/components/TestShowMoreCard.js (hunk 1 above)
Comment (tsl143) [reply tests/unit/ui/components/TestShowMoreCard.js]: This make things easy :)

File: src/ui/components/ShowMoreCard/index.js [hunk 3]
```diff
@@ -11,8 +11,8 @@ import Card from 'ui/components/Card';
 
 import './ShowMoreCard.scss';
 
-
-const MAX_HEIGHT = 100;
+// This refers to height of Card-contents class, beyond which it will add read more link
```
Comment (willdurand) [inline src/ui/components/ShowMoreCard/index.js]: I would write:

    `.Card-contents` CSS class

A period is missing at the end of the sentence, which should be wrapped to 80 chars too.

**Comment by tsl143** [comment]
@wagnerand , updated the comment.
Also if we can put 80 char/line for comment limit in lint rules, that will help.

File: src/ui/components/ShowMoreCard/index.js [hunk 4]
```diff
@@ -11,8 +11,9 @@ import Card from 'ui/components/Card';
 
 import './ShowMoreCard.scss';
 
-
-const MAX_HEIGHT = 100;
+// This refers to height of `Card-contents` CSS class,
+// beyond which it will add read more link.
+export const MAX_HEIGHT = 145;
```
Comment (willdurand) [inline src/ui/components/ShowMoreCard/index.js]: Why not having `150` actually? I cannot get the issue mentioned in #3148 fixed with this value.

File: src/ui/components/ShowMoreCard/index.js [hunk 5]
```diff
@@ -40,7 +40,7 @@ export class ShowMoreCardBase extends React.Component {
   truncateToMaxHeight = (contents) => {
     // If the contents are short enough they don't need a "show more" link; the
     // contents are expanded by default.
-    if (contents.clientHeight > MAX_HEIGHT) {
+    if (contents.clientHeight >= MAX_HEIGHT) {
```
Comment (tsl143) [inline src/ui/components/ShowMoreCard/index.js]: Need this as, when content loads the due to max height the height of content equals 150px, but again on `componentWillReceiveProps`, `truncateToMaxHeight` is invoked so even at 150px we need to set `{ expanded: false }`

File: tests/unit/ui/components/TestShowMoreCard.js [hunk 6]
```diff
@@ -40,7 +40,7 @@ describe(__filename, () => {
 
   it('truncates the contents if they are too long', () => {
     const root = render({ children: 'Hello I am description' });
-    root.truncateToMaxHeight({ clientHeight: 101 });
+    root.truncateToMaxHeight({ clientHeight: MAX_HEIGHT + 1 });
```
Comment (muffinresearch) [inline tests/unit/ui/components/TestShowMoreCard.js]: What is the extra 1px needed for?

Comment (tsl143) [reply tests/unit/ui/components/TestShowMoreCard.js]: @muffinresearch , to test `(contents.clientHeight >= MAX_HEIGHT)` this condition.
Initially we were doing it manually `101 px` but then @willdurand suggested to export the variable.

**Comment by kumar303** [comment]
> Also if we can put 80 char/line for comment limit in lint rules, that will help.

It's going to be a big change. I don't think we have bandwidth to do it right now. https://github.com/mozilla/addons-frontend/issues/3045

**Pull request merged**